scripts/cat/cats.py

```diff
@@ -0,0 +1,940 @@
+                self.former_mentor.append(old_mentor)
+
+    def update_mentor(self, new_mentor=None):
+        if new_mentor is None:
+            # If not reassigning and current mentor works, leave it
+            if self.mentor and self.is_valid_mentor(self.mentor):
+                return
+        old_mentor = self.mentor
+        # Should only have mentor if alive and some kind of apprentice
+        if 'apprentice' in self.status and not self.dead and not self.exiled:
+            # Need to pick a random mentor if not specified
+            if new_mentor is None:
+                potential_mentors = []
+                priority_mentors = []
+                for cat in self.all_cats.values():
+                    if self.is_valid_mentor(cat):
+                        potential_mentors.append(cat)
+                        if len(cat.apprentice) == 0:
+                            priority_mentors.append(cat)
+                # First try for a cat who currently has no apprentices
+                if len(priority_mentors) > 0:
+                    new_mentor = choice(priority_mentors)
+                elif len(potential_mentors) > 0:
+                    new_mentor = choice(potential_mentors)
+            # Mentor changing to chosen/specified cat
+            self.mentor = new_mentor
+            if new_mentor is not None:
+                if self not in new_mentor.apprentice:
+                    new_mentor.apprentice.append(self)
+                if self in new_mentor.former_apprentices:
+                    new_mentor.former_apprentices.remove(self)
+        else:
+            self.mentor = None
+        # Move from old mentor's apps to former apps
+        if old_mentor is not None and old_mentor != self.mentor:
+            if self in old_mentor.apprentice:
+                old_mentor.apprentice.remove(self)
+            if self not in old_mentor.former_apprentices:
+                old_mentor.former_apprentices.append(self)
+            if old_mentor not in self.former_mentor:
+                self.former_mentor.append(old_mentor)
+
+# ---------------------------------------------------------------------------- #
+#                                  properties                                  #
+# ---------------------------------------------------------------------------- #
+
+    @property
+    def experience(self):
+        return self._experience
+
+    @experience.setter
+    def experience(self, exp):
+        if (exp > 80):
+            exp = 80
+        self._experience = exp
+        experience_levels = [
+            'very low', 'low', 'slightly low', 'average', 'somewhat high',
+            'high', 'very high', 'master', 'max'
+        ]
+        self.experience_level = experience_levels[math.floor(self.experience /
+                                                             10)]
+
+    @property
+    def moons(self):
+        return self._moons
+
+    @moons.setter
+    def moons(self, value):
```

Comment:
I get a bug where all my starter cats have their age listed as elder, and an elder sprite, despite being non-elder moons and statuses. Pretty sure it's something to do with this function

Reply:
Yes probably, this function caused in general some problems, maybe I will remove them and add the function check_age again.